crowbar_framework/app/models/deployer_service.rb

```diff
@@ -99,9 +101,11 @@ def transition(inst, name, state)
         break if done
       end
 
-      os_map = role.default_attributes["deployer"]["os_map"]
-      node.crowbar["crowbar"]["hardware"]["os"] = os_map[0]["install_os"]
-      node.save
+      unless node.crowbar["crowbar"]["hardware"].nil?
```

Comment:
Error is here http://pastebin.nue.suse.com/18621/src

It should be fixed mostly by the return of `roles << node.run_list_to_roles`, this extra check for nil is just a failsafe.